Sources/Extensions/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -135,6 +135,27 @@ public extension Sequence {
         })
     }
 
+    /// SwifterSwift: Get the only element based on optional condition.
+    ///
+    ///     [].single() -> nil
+    ///     [4].single() -> 4
+    ///     [2, 4].single() -> nil
+    ///     [1, 4, 7].single(where: {$0 % 2 == 0}) -> 4
+    ///     [2, 2, 4, 7].single(where: {$0 % 2 == 0}) -> nil
+    ///
+    /// - Parameter condition: condition to evaluate each element against (default is `{_ in true}`).
+    /// - Returns: The only element in the array matching the specified condition. If there are more matching elements, nil is returned. (optional)
+    public func single(where condition: ((Element) throws -> Bool) = { _ in true }) -> Element? {
```

Comment:
@LucianoPAlmeida I agree on the throw and the default value part.

I'd personally call the method `unique(where: )`. The `firstIf` is not required because unique should already tell that it can only be "the one and only".

Reply:
Hey @BennX Interesting point you made 👍  
but I think `first` gives the notion that the search for an element that matches the condition is start to end, so it takes the first element that matches :))

Reply:
@LucianoPAImeida But does that matter? The element need to be unique anyways, so the traversation order is irrelevant from the naming perspective.

Reply:
I believe yes because, for example, if we have an array [1, 2, 3, 4, 5] 
and call `single(where: { $0 % 2 == 0 })`, both 2 and 4 satisfy the condition and are unique, but if we aproach start...end  result will be 2 and if we approach end...start will be 4 :))

Reply:
Hey @LucianoPAlmeida and @BennX, thank you for your comments.

* I chose the name, since I got the idea from [LINQ](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single?view=netframework-4.7.2), and there it’s called `Single()`. I also thought about a different name, but I didn’t found a good replacement. But the suggestion `unique(where:)` sounds also good to me.

* In C# there is also the default value.
```c#
string fruit1 = fruits1.Single();
string fruit1 = fruits.Single(fruit => fruit.Length > 10);
```
Actually, I use the default value quite often, when I have a query, from that I only expect one result. So I think, it helps to have it.

* I usually use the method like:
```swift
if let result = queryResult.single() {
   print(result)
}
```
In the case, that it `rethrows`, I need to have a `try?` there, which leads to a double optional. So then it’s not that nice to use. Or am I missing something there?

Reply:
Don't think `unique` is a good name because it makes me think the return value is going to be a boolean. 
I would be ok with something like `firstUnique(where:)`. However, I actually prefer the name `single` if it's being used in C#.

Reply:
@LucianoPAlmeida
It’s true, that `rethrows` follows more the standard. The problem there is, that in that way the default value is not recognised as non-throwing. Therefore I would suggest to remove the default value and add another method, to have a simpler use:
```swift
public func single() -> Element? {
    return single { _ in true }
}
```

Reply:
@andlang We don't allow 1 SwifterSwift method to be dependent on another SwifterSwift method, so we can't make the addition you just described. I definitely think this method needs to follow the `throws` & `rethrows` pattern. If you can't get the default value to compile, I would just drop it.

Reply:
Hey @andlang, @SD10 Sorry the late response, but I agree with other method implementation 👍